Xamarin.PropertyEditing/Extensions.cs

```diff
@@ -30,6 +30,26 @@ public static void AddItems<T> (this ICollection<T> self, IEnumerable items)
 				self.Add (item);
 		}
 
+		public static void ReplaceOrAdd<T> (this ICollection<T> self, T replace, T with)
```

Comment:
Both string and regex versions or `Replace` replace all occurrences, so it may be surprising that this does only the first one. I'd at least add a doc comment specifying it, or rename the method to `ReplaceFirstOrAdd`.